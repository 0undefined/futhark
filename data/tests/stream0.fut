fun {int,[real]} main(*[int] A) = 
  stream( fn {int,*[real]} (int acc, *[int] a) =>
                    let x = map (+4,   a ) in 
                    let y0= scan(+, 0, x ) in
                    let y = map (+acc, y0) in
                    let z = map ( fn real (int a) => toReal(3*a), y) in
                    let u = y0[chunk-1] 
                    in { acc+u, copy(z) }
        , chunk, i, 0, A
        )
