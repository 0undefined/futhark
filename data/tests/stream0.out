{54, [12.0, 30.0, 54.0, 84.0, 120.0, 162.0]}

